chain/epoch_manager/src/validator_selection.rs

```diff
@@ -156,6 +197,8 @@ pub fn proposals_to_epoch_info(
     ))
 }
 
+/// Generates proposals based on new proposals, last epoch validators/fishermen and validator
+/// kickouts
```

Comment:
from what I see it does a lot more (and we should consider splitting this function into separate ones in the future).

it:
- looks through new proposals and reject the ones that should be kicked out
- gives rewards to validators from the previous epoch (and adds them to the validator list of the next epoch)
- it also 'unstakes' all the fishermen that were present in the last epoch